src/backend/distributed/metadata/dependency.c

```diff
@@ -235,6 +238,29 @@ GetAllDependenciesForObject(const ObjectAddress *target)
 }
 
 
+/*
+ * GetAllDependenciesForObject returns a list of all the dependent objects of the given
+ * object irrespective of whether the dependent object is supported by Citus or not.
+ * This function will be used to provide meaningful error messages if any dependent
+ * object for a given object is not supported. If you want to create dependencies for
+ * an object, you probably need to use GetDependenciesForObject().
+ */
+List *
+GetAllDependenciesForObject(const ObjectAddress *target)
+{
+	ObjectAddressCollector collector = { 0 };
+	InitObjectAddressCollector(&collector);
+
+	RecurseObjectDependencies(*target,
+							  &ExpandCitusSupportedTypes,
```

Comment:
I guess to make `GetAllDependenciesForObject` actually cover **ALL** we should not use `ExpandCitusSupportedTypes` but something more generic.

I don't want to get into such a change as it probably wouldn't add much value for our purpose, but we can at least change the comment to reflect this?